client/configuration/search_path.py

```diff
@@ -19,13 +19,22 @@
 import glob
 import logging
 import os
-from typing import Dict, Iterable, List, Sequence, Union
+import re
+from typing import Dict, Iterable, List, Sequence, Tuple, Union
 
 from .. import filesystem
 from . import exceptions
 
 LOG: logging.Logger = logging.getLogger(__name__)
 
+dist_info_in_root: Dict[str, List[str]] = {}
+
+# pyre-fixme[5]: Globally accessible variable `_site_filter` has type `re.Pattern[str]` but no type is specified.
+_site_filter = re.compile(r".*-([0-99]\.)*dist-info")
+
+# pyre-fixme[5]: Globally accessible variable `_PYCACHE` has type `re.Pattern[str]` but no type is specified.
+_PYCACHE = re.compile("__pycache__(/)*.*")
```

Comment:
No annotation for these two vars to keep compatible with Python3.8

Reply:
could we at least type it as re.Pattern (without the generic)? otherwise, could we put a comment here linking to the docs and explaining why the type is omitted? https://docs.python.org/3/library/re.html#re.Pattern

Reply:
> could we at least type it as re.Pattern (without the generic)? otherwise, could we put a comment here linking to the docs and explaining why the type is omitted? https://docs.python.org/3/library/re.html#re.Pattern

`re.Pattern[str]` failed in Python3.8.
https://github.com/facebook/pyre-check/actions/runs/5922502819/job/16056564658#step:5:107

Never mind, I specifed type by comment.

Reply:
you also might be able to try different annotations in each branch of a `if sys.version_info <= ...` check

Reply:
But specify type by comment is work in all the version of Python, why do I need to write special case for 3.8?